### PR TITLE
Create job from cron job

### PR DESCRIPTION
**Pull request by gobozov**
Creating job from cronjob. Based on kubectl source code https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/create/create_job.go
Original issue https://github.com/kubernetes/kubernetes/issues/47538

### CODE REVIEW

**Comment by gobozov** [comment]
I signed it

File: kubernetes/src/main/java/io/kubernetes/client/openapi/ModelUtil.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package io.kubernetes.client.openapi;
```
Comment (brendandburns) [inline kubernetes/src/main/java/io/kubernetes/client/openapi/ModelUtil.java:1]: This should be in the `util/src/main/java/...` directory, this directory is all generated code.

File: kubernetes/src/main/java/io/kubernetes/client/openapi/apis/BatchV1Api.java [hunk 2]
```diff
@@ -13,26 +13,21 @@
 
 package io.kubernetes.client.openapi.apis;
 
+import com.google.gson.reflect.TypeToken;
```
Comment (brendandburns) [inline kubernetes/src/main/java/io/kubernetes/client/openapi/apis/BatchV1Api.java:16]: This code is all generated, we don't edit it in this project.

**Comment by brendandburns** [comment]
Most of the code in this library is code generated. We can't modify it because it will prevent future code generation runs.

If you want to add this, you need to add it as a separate class in the `util` directory, similar to how `Exec` and `Attach` etc were added.

File: kubernetes/src/main/java/io/kubernetes/client/openapi/ModelUtil.java (hunk 1 above)
Comment (yue9944882) [reply kubernetes/src/main/java/io/kubernetes/client/openapi/ModelUtil.java:1]: would rename the package path to `io.kubernetes.client.util.conversion.Jobs`

Comment (gobozov) [reply kubernetes/src/main/java/io/kubernetes/client/openapi/ModelUtil.java:1]: ok, I didn't figure out project structure. I can move into util

File: kubernetes/src/main/java/io/kubernetes/client/openapi/apis/BatchV1Api.java (hunk 2 above)
Comment (gobozov) [reply kubernetes/src/main/java/io/kubernetes/client/openapi/apis/BatchV1Api.java:16]: ok, where all api methods gets defined?

Comment (gobozov) [reply kubernetes/src/main/java/io/kubernetes/client/openapi/apis/BatchV1Api.java:16]: Just trying to understand... You mean we can not change api methods for separate project aka adding BatchV1Api.createNamespacedJobFromCronJob, we can add util method for conversion, call it by passing cronJob object and pass result to BatchV1Api.createNamespacedJob to run job. Correct?

Comment (brendandburns) [reply kubernetes/src/main/java/io/kubernetes/client/openapi/apis/BatchV1Api.java:16]: Yes, that's correct. Basically, we don't edit things in the `kubernetes` directory (mostly, there is a `custom` dir in there...)

Comment (gobozov) [reply kubernetes/src/main/java/io/kubernetes/client/openapi/apis/BatchV1Api.java]: Examples self generated as well? Wonder if I can add example class to show how this work.

**Comment by brendandburns** [comment]
CI/CD looks like it got disabled. We need to switch to github actions anyway, I need to hold off merging this until we get CI/CD fixed.

**Comment by brendandburns** [comment]
close/reopen to trigger CI/CD

**Comment by yue9944882** [comment]
> [ERROR] Non complying file: /home/runner/work/java/java/util/src/main/java/io/kubernetes/client/util/conversion/Jobs.java
> [ERROR] Failed to execute goal com.coveo:fmt-maven-plugin:2.5.0:check (default) on project client-java: Found 1 non-complying files, failing build -> [Help 1]

@gobozov the CI job is failing on format-check, plz run the plugin to fix it

**Comment by gobozov** [comment]
Fixed formatting.

**Comment by gobozov** [comment]
How to get this merged? I see that approved label is required.

**Comment by brendandburns** [comment]
/approve

Sorry for the delay

**Comment by tozka** [comment]
Hi, 

How do I see in which k8s release is this change going to be part of ?

**Pull request merged**